src/index.js

```diff
@@ -1,16 +1,170 @@
+    } else {
+      str = resultFromCourseValidation.str;
+      courseNumber = resultFromCourseValidation.courseNumber;
+      departmentName = resultFromCourseValidation.departmentName;
+      i = resultFromCourseValidation.i;
+    }
+    // Check if space between Course and Semester Strings occur
+    if (i >= str.length) {
+      return {
+        message:
+          "Semester Information missing after Department+Course Information",
+        error: true
+      };
+    }
+    if (str[i++] !== " ") {
+      return {
+        message:
+          "Space required between Department+Course info and Semester+Year info",
+        error: true
+      };
+    }
+    // check if invalid semester+year info i.e. either letter or digit
+    if (CommonUtils.isLetter(str[i]) === false && isNaN(str[i]) === true) {
+      return {
+        message: "Invalid character in Semester+Year info with: " + str[i],
+        error: true
+      };
+    }
+
+    // console.log(SemesterUtils);
+
+    let resultFromSemesterValidation = SemesterUtils.validateSemesterData(
+      str,
+      semesterName,
+      year,
+      i
+    );
+
+    if (resultFromSemesterValidation.error === true) {
+      return resultFromSemesterValidation;
+    } else {
+      str = resultFromSemesterValidation.str;
+      semesterName = resultFromSemesterValidation.semesterName;
+      year = resultFromSemesterValidation.year;
+      i = resultFromSemesterValidation.i;
+    }
+
+    // (// console.log("departmentName:" + departmentName);
+    // console.log("course number:" + courseNumber);
+    // console.log("semName:" + semesterName);
+    // console.log("year:" + year);
+    TABLEROWS = [
+      createData("Department:", departmentName),
+      createData("Course Number:", courseNumber),
+      createData("Semester:", semesterName),
+      createData("Year:", year)
+    ];
+
+    return {
+      message: "SUCCESS",
+      error: false
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <label>
+          Name:
+          <input type="text" ref={this.inputRef} />
+        </label>
+        <input type="submit" onClick={this.processInput} />
+        <br />
+        <label style={{ color: this.state.resultObj.error ? "red" : "green" }}>
+          {this.state.resultObj.error ? (
+            this.state.resultObj.message
+          ) : (
+            <div>
+              <br />
+              <Paper classes={useStyles.root}>
+                <Table classes={useStyles.table}>
+                  <TableBody>
+                    {TABLEROWS.map(row => (
+                      <TableRow key={row.name}>
+                        <TableCell component="th" scope="row">
+                          {row.name}
+                        </TableCell>
+                        <TableCell align="right">{row.value}</TableCell>
+                      </TableRow>
+                    ))}
+                  </TableBody>
+                </Table>
+              </Paper>
+            </div>
```

Comment:
After above comment is resolved, unconditionally display             this.state.resultObj.message